.github/workflows/build_test_template.yml

```diff
@@ -0,0 +1,70 @@
+name: Build Template
+
+on:
+  workflow_call:
+    secrets:
+      E2B_TESTS_ACCESS_TOKEN:
+        required: true
+    inputs:
+      E2B_DOMAIN:
+        required: false
+        type: string
+    outputs:
+      template_id:
+        description: "The ID of the built template"
+        value: ${{ jobs.build.outputs.template_id }}
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Build Docker Image and Template
+    runs-on: ubuntu-latest
+    outputs:
+      template_id: ${{ steps.build-template.outputs.template_id }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set package version
+        working-directory: ./template
+        run: |
+          VERSION=$(cat ../chart_data_extractor/pyproject.toml | grep version | cut -d '"' -f 2)
+          echo "Version: $VERSION"
+          sed -i "s/e2b_charts/e2b_charts==${VERSION}/g" requirements.txt
+
+      - name: Build Docker image
+        id: docker-build
+        uses: docker/build-push-action@v6
+        with:
+          context: ./template
+          push: false
+          platforms: linux/amd64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          tags: e2bdev/code-interpreter-test:latest
+
+      - name: Install E2B CLI
+        run: npm install -g @e2b/cli
+
+      - name: Build E2B template
+        id: build-template
+        run: |
+          rm -f e2b.toml
+          echo "FROM e2bdev/code-interpreter:latest" > test.Dockerfile
+          e2b template build --memory-mb 1024 -c "/root/.jupyter/start-up.sh" -d "test.Dockerfile" | tee build_output.txt
+          TEMPLATE_ID=$(grep "Requested build for the sandbox template" build_output.txt | awk '{print $NF}')
```

Comment:
Can you make sure this fail if the build fails. Could you read the template id from `e2b.toml`?

Reply:
on it